src/Components/OATHeader/internal/FormSettings.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
+import {
+    TextField,
+    Text,
+    ActionButton,
+    FontIcon,
+    PrimaryButton,
+    Stack
+} from '@fluentui/react';
+import { useTranslation } from 'react-i18next';
+import { IAction } from '../../../Models/Constants/Interfaces';
+import { SET_OAT_NAMESPACE } from '../../../Models/Constants/ActionTypes';
+import { getHeaderStyles } from '../OATHeader.styles';
+import { DTDLNameRegex } from '../../../Models/Constants/Constants';
+import { IOATEditorState } from '../../../Pages/OATEditorPage/OATEditorPage.types';
+
+interface IModal {
+    dispatch?: React.Dispatch<React.SetStateAction<IAction>>;
+    setModalOpen?: React.Dispatch<React.SetStateAction<boolean>>;
+    setModalBody?: React.Dispatch<React.SetStateAction<string>>;
+    state?: IOATEditorState;
+}
+
+export const FormSettings = ({
+    setModalOpen,
+    dispatch,
+    setModalBody,
+    state
+}: IModal) => {
+    const { t } = useTranslation();
+    const [namespace, setNamespace] = useState(state.namespace);
+    const [error, setError] = useState(false);
+    const headerStyles = getHeaderStyles();
+
+    const handleProjectNamespaceChange = (value) => {
+        //  The name may only contain the characters a-z, A-Z, 0-9, and underscore.
+        if (DTDLNameRegex.test(value)) {
```

Comment:
If this limit is correct, should be localizing the default value?

Reply:
shoiuld be good to go @jamescarpinter